diff -ruN anthy-9100b.orig/mkworddic/dict.args.in anthy-9100b/mkworddic/dict.args.in
--- anthy-9100b.orig/mkworddic/dict.args.in	2007-09-16 21:38:40.000000000 +0900
+++ anthy-9100b/mkworddic/dict.args.in	2007-09-18 19:53:52.000000000 +0900
@@ -10,9 +10,9 @@
 read @top_srcdir@/alt-cannadic/gcanna.ctd
 #
 read @top_srcdir@/alt-cannadic/gcannaf.ctd
-#read @top_srcdir@/alt-cannadic/gtankan.ctd
+read @top_srcdir@/alt-cannadic/gtankan.ctd
 #
-read @top_srcdir@/mkworddic/gtankan.ctd
+#read @top_srcdir@/mkworddic/gtankan.ctd
 read @top_srcdir@/mkworddic/adjust.t
 read @top_srcdir@/mkworddic/name.t
 read @top_srcdir@/mkworddic/compound.t
