diff -pruN anthy-9100h.orig/src-diclib/xstr.c anthy-9100h/src-diclib/xstr.c
--- anthy-9100h.orig/src-diclib/xstr.c	2009-01-23 12:31:31.000000000 +0900
+++ anthy-9100h/src-diclib/xstr.c	2013-06-17 19:30:08.000000000 +0900
@@ -384,6 +384,10 @@ int
 anthy_xstrcmp(xstr *x1, xstr *x2)
 {
   int i, m;
+  if (!x1)
+    return -1;
+  if (!x2)
+    return 1;
   if (x1->len < x2->len) {
     m = x1->len;
   }else{
